File: PNJGL_onweb/data_parameters.m
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% ------------------- SYNTHETIC DATA GENERATION PARAMETERS ------------------%
%									     %	
% LAST UPDATE: 3/1/2013 						     %	
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%



% ----------- DATA generation parameters
umin_sparse = 0.3; % Lower and upper bounds for magnitude of entries for the sparse matrix
umax_sparse = 0.6;
umin_pert = 0.3; % Lower and upper bounds for magnitude of perturbations
umax_pert = 0.6;
sparsity_prob = 0.9; % Sparsity level desired in the true inverse covariance matrix
 

percentage_perturbed = 2;
m_pert = ceil((percentage_perturbed/100)*p); % Number of perturbed nodes

File: Python/data_parameters.py
from math import ceil

# Lower and upper bounds for magnitude of entires for the sparse matrix
umin_sparse = 0.3
umax_sparse = 0.6

# Lower and upper bounds for magnitude of entires in common cohub nodes
umin_common = 0.3
umax_common = 0.6

# Lower and upper bounds for magnitude of perturbations
umin_pert = 0.3
umax_pert = 0.6

# Sparsity level desired in the true inverse covariance matrix
sparsity_prob = 0.98


percentage_common = 2
percentage_pert = 1

# Number of common cohub nodes
def m_common(p):
    ceil((percentage_common/100)*p)

# Number of perturbed nodes
def m_pert(p):
    ceil((percentage_pert/100)*p)
